Common/Securities/MarketHoursDatabase.cs

```diff
@@ -237,5 +237,10 @@ public override Entry GetEntry(string market, string symbol, SecurityType securi
                 return new Entry(tz, SecurityExchangeHours.AlwaysOpen(tz));
             }
         }
+
+        public void AddEntry(SecurityDatabaseKey key, Entry entry)
+        {
+            _entries[key] = entry;
```

Comment:
This should be `_entries.Add` so that we get an exception when adding an entry that already exists. This also means that consumers who call add should first try to get a matching, symbol specific entry -- perhaps a `Contains` method is needed to make that easier for consumers?